codec/src/test/java/io/netty/handler/codec/DatagramPacketEncoderTest.java

```diff
@@ -82,4 +86,45 @@ public void testUnmatchedType() {
         assertTrue(channel.writeOutbound(netty));
         assertSame(netty, channel.readOutbound());
     }
+
+    @Test
+    public void testNotSharable() {
+        MessageToMessageEncoder<AddressedEnvelope<ByteBuf, InetSocketAddress>> wrapped =
+                new TestMessageToMessageEncoder(false);
+        DatagramPacketEncoder<AddressedEnvelope<ByteBuf, InetSocketAddress>> encoder =
+                new DatagramPacketEncoder<AddressedEnvelope<ByteBuf, InetSocketAddress>>(wrapped);
+        assertEquals(wrapped.isSharable(), encoder.isSharable());
+    }
+
+    @Test
+    public void testSharable() {
+        MessageToMessageEncoder<AddressedEnvelope<ByteBuf, InetSocketAddress>> wrapped =
```

Comment:
nit this can be extracted out into a shared method:

```java
private void testSharable(boolean sharable) {
  MessageToMessageEncoder<AddressedEnvelope<ByteBuf, InetSocketAddress>> wrapped =
    new TestMessageToMessageEncoder(sharable);
  DatagramPacketEncoder<AddressedEnvelope<ByteBuf, InetSocketAddress>> encoder =
    new DatagramPacketEncoder<AddressedEnvelope<ByteBuf, InetSocketAddress>>(wrapped);
  assertEquals(wrapped.isSharable(), encoder.isSharable());
}
```